packages/patternfly-4/react-core/src/components/Pagination/Navigation.tsx

```diff
@@ -40,97 +41,148 @@ export interface NavigationProps extends React.HTMLProps<HTMLElement> {
-      aria-label={toFirstPage}
-      data-action="first"
-      onClick={event => {
-        onFirstClick(event, 1);
-        onSetPage(event, 1);
-      }}
-    >
-      <AngleDoubleLeftIcon />
-    </Button>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === 1}
-      data-action="previous"
-      onClick={event => {
-        const newPage = page - 1 >= 1 ? page - 1 : 1;
-        onPreviousClick(event, newPage);
-        onSetPage(event, newPage);
-      }}
-      aria-label={toPreviousPage}
-    >
-      <AngleLeftIcon />
-    </Button>
-    <div className={css(styles.paginationNavPageSelect)}>
-      <input
-        className={css(styles.formControl)}
-        aria-label={currPage}
-        type="number"
-        min="1"
-        max={lastPage}
-        value={page}
-        onChange={event => {
-          let inputPage = Number.parseInt(event.target.value, 10);
-          inputPage = Number.isNaN(inputPage) ? page : inputPage;
-          inputPage = inputPage > lastPage ? lastPage : inputPage;
-          inputPage = inputPage < 1 ? 1 : inputPage;
-          onSetPage(event, Number.isNaN(inputPage) ? page : inputPage);
-          onPageInput(event, Number.isNaN(inputPage) ? page : inputPage);
-        }}
-      />
-      <span aria-hidden="true">
-        of {pluralize(lastPage, pagesTitle)}
-      </span>
-    </div>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === lastPage}
-      aria-label={toNextPage}
-      data-action="next"
-      onClick={event => {
-        const newPage = page + 1 <= lastPage ? page + 1 : lastPage;
-        onNextClick(event, newPage);
-        onSetPage(event, newPage);
-      }}
-    >
-      <AngleRightIcon />
-    </Button>
-    <Button
-      variant={ButtonVariant.plain}
-      isDisabled={page === lastPage}
-      aria-label={toLastPage}
-      data-action="last"
-      onClick={event => {
-        onLastClick(event, lastPage);
-        onSetPage(event, lastPage);
-      }}
-    >
-      <AngleDoubleRightIcon />
-    </Button>
-  </nav>
-);
+export interface NavigationState {
+  userInputPage?: React.ReactText | number;
+}
+
+export class Navigation extends React.Component<NavigationProps, NavigationState> {
+  constructor(props: NavigationProps) {
+    super(props);
+    this.state = { userInputPage: this.props.page };
+  }
+
+  static defaultProps = {
+    className: '',
+    lastPage: 0,
+    pagesTitle: '',
+    toLastPage: 'Go to last page',
+    toNextPage: 'Go to next page',
+    toFirstPage: 'Go to first page',
+    toPreviousPage: 'Go to previous page',
+    currPage: 'Current page',
+    paginationTitle: 'Pagination',
+    onNextClick: () => undefined as any,
+    onPreviousClick: () => undefined as any,
+    onFirstClick: () => undefined as any,
+    onLastClick: () => undefined as any,
+    onPageInput: () => undefined as any,
+  };
+
+  private parseInteger(input: React.ReactText, lastPage: number): number | string {
```

Comment:
Make this function `static` since it doesn't depend on `this`. Also, return type should be just `number`, I don't think there's a way this ever returns a `string`.

Reply:
@tlabaj - Are you ok with making it static instead of private?

Reply:
I think it can be both! `static private parseInteger`...